### ISSUE TITLE
Review imagePullPolicy setttings for releases

### ISSUE THREAD

**Issue by youngnick** [projectcontour/contour#1528]
Currently, the imagePullPolicy for the example deployment is set to `Always`. For releases, this is probably not necessary, since the image will not change very regularly.

This issue is to reexamine how we generate release examples, and if we should change this behaviour.

**Comment by davecheney** [comment]
@youngnick you raised this one a while back. Would you be able to look into it this week?

**Comment by youngnick** [comment]
I think that, given we now have the example deployment using the tagged release version, from a saving-bandwidth perspective, it's better to have the ImagePullPolicy for Contour and Envoy pods be `IfNotPresent`.

There is a valid security concern in the more general case about people being able to use the cached version of an image when they are not allowed to pull it (for confidential images). However, given that both Contour and Envoy are open-source images, I don't see that being a problem.

In addition, there is a (small, but present) security risk that someone could mutate the on-disk container image in a bad way, which would then mean that a bad image would be loaded. Again, I don't think this applies to us as much because this is an open-source image, and we are reasonably cautious about RBAC for the app itself.

tl;dr I'll change the policy to `IfNotPresent` now.

**Comment by youngnick** [comment]
There was a slight complication to this one: this needs to only be done in the rendered yaml, as that's what we change for each release.
